cli/cmd/cluster/create.go

```diff
@@ -1,43 +1,148 @@
 package cluster
 
 import (
-	"errors"
 	"fmt"
+	"os"
 
 	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	errorsAgg "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/grtl/mysql-operator/cli/config"
+	"github.com/grtl/mysql-operator/cli/options"
+	mysqlv1 "github.com/grtl/mysql-operator/pkg/apis/cr/v1"
 )
 
 var (
-	storage    string
-	backupName string
+	replicas       int32
+	password       string
+	storage        string
+	secretName     string
+	backupName     string
+	backupInstance string
+	fromSecret     string
+	port           int32
+	image          string
 )
 
 var clusterCreateCmd = &cobra.Command{
 	Use:   "create [cluster name]",
-	Short: "short description of cluster create",
-	Long: `A longer description of cluster create with examples:
-msp cluster create "my-cluster" --storage 1Gi`,
-	Args: func(cmd *cobra.Command, args []string) error {
-		if len(args) < 1 {
-			return errors.New("Cluster name is required")
-		}
-		return nil
-	},
+	Short: "Creates MySQL cluster",
+	Long: `Creates new MySQL cluster.
+You can specify your own secret using from-secret flag:
+msp cluster create "my-cluster" --from-secret "my-secret"
+or create new secret using secret and password flags:
+msp cluster create "my-cluster" --secret "your-new-secret" --password "mysql-password"
+In order to create cluster from backup use backup and instance flags:
+msp cluster create "my-cluster" --backup "backup-name" --instance "instance"`,
+	Args: cobra.MinimumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		if backupName != "" {
-			//TODO: create cluster from backup logic
-			fmt.Println("create cluster --storage --from-backup called")
+		options := options.ExtractOptions(cmd)
+
+		if fromSecret == "" {
+			if secretName == "" || password == "" {
+				fmt.Fprintln(os.Stderr, "You must provide secret and password (or create"+
+					" cluster from existing secret using from-secret flag).")
+				fmt.Fprintln(os.Stderr, "Type msp help cluster create for more information.")
+				os.Exit(1)
+			}
+			err := createSecret(options)
+			if err != nil {
```

Comment:
This construct is used quite often - maybe add another function to the utils library
```go
func FailWithError(err error) {
    fmt.Fprintln(os.Stderr, err.Error())
    os.Exit(1)
}
```